Flip autoconnected slots to explicit connections, pt 1

nyalldawson opened the pull request:
Using https://phabricator.kde.org/D8187, this PR flips a bunch of autoconnected slots to explicit connections.

Why?

Because Qt autoconnected slots were a bad idea. They are extremely fragile and give no build or run time errors when the connection is broken. They make refactoring dialogs a pain. They don't work with IDEs when you search for uses of a widget.

Basically, s*** all round. Let's get rid of them and have a better future.

In fact, they're such a bad idea, I suspect Qt will drop them sometime. Better to dump them before they do.

nyalldawson committed:
Flip autoconnected slots to explicit connections, pt 1

m-kuhn commented on python/gui/qgsrasterformatsaveoptionswidget.sip:
```diff
@@ -104,17 +104,17 @@ class QgsRasterFormatSaveOptionsWidget: QWidget
 
   private slots:
-    void on_mProfileNewButton_clicked() ;
+    void mProfileNewButton_clicked() ;
```
Wow, that's a lot of `private slots` here :D

nyalldawson replied:
Hmm... yes.... @3nids any idea why these are sneaking into the bindings?

3nids replied:
@nyalldawson should be fixed by #5328

m-kuhn reviewed (approved):
Yes, please

nyalldawson committed:
Remove autoconnected slots, pt 2

nyalldawson committed:
Remove all previously autoconnected slots from public API

These are implementation details only, and should not be
part of the stable API.

We need the flexibility to change these at any stage without
being bound by any API contract.

nyalldawson committed:
Fix missing connection

nyalldawson committed:
Make more slots private
